### PR TITLE
Add support for upgrading kafka clusters

### PR DESCRIPTION
**Pull request by halorgium**
This adds support for upgrading a kafka cluster to a specified version.
Not all versions are supported.

### CODE REVIEW

File: commands/clusters.js [hunk 1]
```diff
@@ -63,6 +63,22 @@ HerokuKafkaClusters.prototype.fail = function* (cluster, catastrophic, zk) {
   }
 };
 
+HerokuKafkaClusters.prototype.upgrade = function* (cluster, flags) {
+  var addon = yield this.addonForSingleClusterCommand(cluster);
+  if (addon) {
+    var response = yield this.request({
+      method: 'PUT',
+      body: {
+        version: flags.version
```
Comment (uhoh-itsmaciek) [inline commands/clusters.js]: Since `version` is the only thing we want from the flags, maybe pass that in directly?

File: commands/upgrade.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+let FLAGS = [
+  {name: 'version',
+    description: 'requested kafka version for upgrade',
+    hasValue: true},
+  {name: 'confirm',
```
Comment (uhoh-itsmaciek) [inline commands/upgrade.js]: There's a `.confirm` in `heroku-cli-util` that we can use now: https://github.com/heroku/heroku-cli-util#confirm-app -- I've been meaning to move our other usage to this for consistency with other plugins.

File: commands/upgrade.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+let co = require('co');
+let HerokuKafkaClusters = require('./clusters.js').HerokuKafkaClusters;
+let parseDuration = require('./shared').parseDuration;
+let sleep = require('co-sleep');
+let _ = require('underscore');
+
+function extractFlags(contextFlags) {
+  // This just ensures that we only ever get the flags we expect,
+  // and don't get any additional keys out of the heroku cli flags object
+  // (if there happen to be any).
```
Comment (uhoh-itsmaciek) [inline commands/upgrade.js]: This is obsolete--the CLI takes care of that:

```console
maciek@mothra:~$ heroku kafka:info --not-a-real-flag --app maru-maciek
 ▸    Error: Unexpected flag --not-a-real-flag
 ▸    Usage: heroku kafka:info [CLUSTER]
 ▸    
 ▸    This flag is invalid for this command. Here are the accepted flags:
 ▸     -a, --app APP       # app to run command against
 ▸     -r, --remote REMOTE # git remote of app to run command against
 ▸    
 ▸    See more information with heroku kafka:info --help
```

File: commands/upgrade.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  var clusters = new HerokuKafkaClusters(heroku, process.env, context);
+  var addon = yield clusters.addonForSingleClusterCommand(context.args.CLUSTER);
+
+  if (!addon) {
+    process.exit(1);
+  }
+
+  var flags = extractFlags(context.flags);
+
+  console.log('flags', JSON.stringify(flags));
```
Comment (uhoh-itsmaciek) [inline commands/upgrade.js]: If you want to keep this in, maybe [`cli.debug`](https://github.com/heroku/heroku-cli-util#debug)?

**Comment by uhoh-itsmaciek** [comment]
Specs fail and some minor comments but other than that +1

File: commands/clusters.js (hunk 1 above)
Comment (halorgium) [inline commands/clusters.js]: into the function? i can do that.

File: commands/upgrade.js (hunk 2 above)
Comment (halorgium) [inline commands/upgrade.js]: ahh, thanks!

File: commands/upgrade.js (hunk 3 above)
Comment (halorgium) [inline commands/upgrade.js]: haha, very good. i felt dirty copying that code.

File: commands/upgrade.js (hunk 4 above)
Comment (halorgium) [inline commands/upgrade.js]: ah, thanks. that isn't needed.

File: commands/upgrade.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  }
+
+  if (context.args.CLUSTER) {
+    process.stdout.write(`Upgrading ${context.args.CLUSTER} to version ${context.flags.version}`);
+  } else {
+    process.stdout.write(`Upgrading to version ${context.flags.version}`);
+  }
+
+  var upgrade = clusters.upgrade(context.args.CLUSTER, context.flags.version);
+  yield printWaitingDots();
```
Comment (uhoh-itsmaciek) [inline commands/upgrade.js]: For what it's worth, I think we may be able to use the CLI support @dickeyxxx is using in `pg:wait` [here](https://github.com/heroku/heroku-pg/pull/6/files#diff-5d23aa2d2daa9961896aa2b7414e0717R31): `cli.action.start` and `cli.action.done`, but what you have is fine for now; we can clean that up later.

File: commands/upgrade.js (hunk 2 above)
Comment (halorgium) [inline commands/upgrade.js]: i'm gonna give up on that path. the upgrade of `heroku-cli-util` exposed some other API changes.

**Comment by uhoh-itsmaciek** [comment]
Nice, ship it.

**Pull request merged**